src/typebox.ts

```diff
@@ -39,6 +41,20 @@ export type TReadonlyOptional<T extends TSchema> = T & { modifier: typeof Readon
 export type TOptional<T extends TSchema>         = T & { modifier: typeof OptionalModifier }
 export type TReadonly<T extends TSchema>         = T & { modifier: typeof ReadonlyModifier }
 
+// --------------------------------------------------------------------------
+// Nominal Types
+// --------------------------------------------------------------------------
+export type BrandType                                 = string | symbol
+export type Brand<BaseType, Brand extends BrandType>  = BaseType & {
+  readonly [__BaseType]: BaseType
+  readonly [__Brand]: Brand
+}
+type Flavor<BaseType, Brand extends BrandType>        = BaseType & {
+  readonly [__BaseType]?: BaseType | undefined
+  readonly [__Brand]?: Brand | undefined
+}
+type StringFlavor                                     = Flavor<string, BrandType>
```

Comment:
We don't export flavors because, in the context of TypeBox, there's no use on making them accessible. We need them, though, to ease the implementation of internal details.

As before, I'm completely open to discuss the names of these types.

Reply:
TypeBox currently provides no facilities for extending types, instead opting for `Type.Intersect([Base, Derived])` which is inline with the structural interpretation only. Not sure if TypeBox can support a base type of this kind (I think there would need to be proper nominal typing support in TypeScript first, as well as some way to adequately capture the relationship between the `Base` and `Derived` type in JSON schema.